src/qibolab/_core/instruments/qcs.py

```diff
@@ -0,0 +1,343 @@
+                            and hw_demod
+                        ):
+                            program = program.n_shots(num_shots)
+                            hw_demod = False
+                elif sweeper.parameter in [
+                    Parameter.amplitude,
+                    Parameter.duration,
+                    Parameter.relative_phase,
+                ]:
+                    # Ignore hardware sweeping if duration is swept
+                    if sweeper.parameter is Parameter.duration and hw_demod:
+                        program = program.n_shots(num_shots)
+                        hw_demod = False
+                    for pulse in sweeper.pulses:
+                        if pulse.id not in sweeper_pulse_map:
+                            sweeper_pulse_map[pulse.id] = {}
+                        sweeper_pulse_map[pulse.id][
+                            sweeper.parameter.name
+                        ] = qcs_variable
+                else:
+                    raise ValueError(
+                        "Sweeper parameter not supported", sweeper.parameter.name
+                    )
+
+                sweep_variables.append(qcs_variable)
+                sweep_values.append(
+                    qcs.Array(
+                        name=f"A{idx}_{idx2}",
+                        value=(
+                            sweeper.values * NANOSECONDS
+                            if sweeper.parameter is Parameter.duration
+                            else sweeper.values
+                        ),
+                        dtype=float,
+                    )
+                )
+            program = program.sweep(sweep_values, sweep_variables)
+
+        # WAVEFORM COMPILATION
+        # Iterate over channels and convert qubit pulses to QCS waveforms
+        for channel_id in sequence.channels:
+            channel = self.channels[channel_id]
+            virtual_channel = self.virtual_channel_map[channel_id]
+
+            if isinstance(channel, AcquisitionChannel):
+                probe_channel_id = channel.probe
+                probe_virtual_channel = self.virtual_channel_map[probe_channel_id]
+                if probe_channel_id in sweeper_channel_map:
+                    frequency = sweeper_channel_map[probe_channel_id]
+                else:
+                    frequency = configs[probe_channel_id].frequency
+                for pulse in sequence.channel(channel_id):
+                    sweep_param_map = sweeper_pulse_map.get(pulse.id, {})
+
+                    if pulse.kind == "delay":
+                        qcs_pulse = qcs.Delay(
+                            sweep_param_map.get(
+                                "duration", pulse.duration * NANOSECONDS
+                            )
+                        )
+                        program.add_waveform(qcs_pulse, virtual_channel)
+                        program.add_waveform(qcs_pulse, probe_virtual_channel)
```

Comment:
So apparently both @stavros11 and I have been adding our delays on the measure channel alone and not including the probe channel
https://github.com/qiboteam/qibocal/blob/0.2/src/qibocal/protocols/rabi/utils.py#L266-L280

Reply:
Just to make you aware: this is working because even @stavros11 is explicitly relying on no one actually using discrete `Acquisition` operations, but only `Readout` operations (to be interpreted as an `Acquisition` + `Pulse`).

This was added because having packaged readouts is the most common workflow. However, if you also plan to use the two channels independently (which should not happen during most protocols, and for sure not during circuit executions), then you may incur in some issues.

In any case, as you noticed, that is also the case of QM (mostly because the QUA API does not support the two operations separately), so if you decide to keep as it would be also fine (and it is unlikely that any user will notice for a long while).